NumLib/DOF/LocalToGlobalIndexMap.h

```diff
@@ -174,6 +189,17 @@ class LocalToGlobalIndexMap final
         std::vector<MeshLib::Element*> const& elements,
         NumLib::MeshComponentMap&& mesh_component_map);
 
+    /// Private constructor used by internally created local-to-global index
+    /// maps. The mesh_component_map is passed as argument instead of being
+    /// created by the constructor.
+    /// \attention The passed mesh_component_map is in undefined state after
+    /// this construtor.
+    explicit LocalToGlobalIndexMap(
+        std::vector<std::unique_ptr<MeshLib::MeshSubsets>>&& mesh_subsets,
+        std::vector<std::size_t> const global_component_ids,
```

Comment:
int for local ids and std::size_t for global.
const& is correct. Now I understand your point about pass-by-value a little better; Those who pass-by-value must std::move that thing, otherwise it's a potential problem.